scripts/environment/gcloud_env.py

```diff
@@ -0,0 +1,219 @@
+#
+# This has been tested with python 2.7.
+
+import re
+import time
+
+from distutils.spawn import find_executable
+from subprocess import PIPE, Popen, call
+
+class GcloudEnvironment(object):
+    """
+    Encapsulate the gcloud setup in the GcloudEnvironment class
+    """
+
+    PROJECT_ID_REGEX = re.compile('^[a-z][a-z0-9-]{6,30}$')
+
+    def __init__(self):
+        self.config_name = None
+        self.auth_account = None
+        self.project_id = None
+        self.region = None
+
+    def __repr__(self):
+        return ('GcloudEnvironment:\n'
+                '  Account: {}\n'
+                '  ProjectId: {}\n'
+                .format(self.auth_account,
+                        self.project_id))
+
+    def ensure_installed(self):
+        """
+        Check whether gcloud tool is installed.
+        """
+        gcloud_cmd = find_executable('gcloud')
+        if gcloud_cmd:
+            print 'Found gcloud tool!'
+        else:
+            raise EnvironmentError(
+                'Could not find gcloud. '
+                'Have you installed the Google Cloud SDK?\n'
+                'You can get it here: https://cloud.google.com/sdk/')
+
+    def auth_login(self):
+        """
+        User needs to authenticate with Google Cloud Platform account
+        before doing anything else.
+        """
+        return_val = call(['gcloud', 'auth', 'login', '--force'])
+        p = Popen(['gcloud', 'auth', 'list',
+                   '--filter=status:ACTIVE', '--format=value(account)'],
+                  stdout=PIPE, stderr=PIPE)
+        stdout, stderr = p.communicate()
+        try:
+            self.account = stdout.strip()
+        except:
+            print 'Invalid account, something went wrong'
+            raise
+
+    def create_or_use_project(self):
+        """
+        Create a project or enter the id of a project to use.
+        """
+        project_id = None
+
+        while True:
+            project_choice = raw_input('Which project do you want to use?\n'
+                                       '[1] Enter a project id\n'
+                                       '[2] Create a new project\n').strip()
+            if project_choice == '1':
+                project_id = self._use_project()
+                break
+            elif project_choice == '2':
+                project_id = self._create_project()
+                break
+        self._set_config(project_id)
+
+    def _create_project(self):
+        """
+        Create the project based on user's input.
+        """
+        while True:
+            project_id = raw_input(
+                'Enter a project id '
+                '(alphanumeric and hyphens): ').strip()
+            if self.PROJECT_ID_REGEX.match(project_id):
+                return_val = call(['gcloud', 'alpha', 'projects', 'create',
+                                  project_id])
+                if not return_val:
+                    return project_id
+                    break
```

Comment:
This is a little confusing due to exit status -- we want to return the project_id if the exit status is 0 (command ran successfully) but not when it fails (exit status 1)